Code/Tools/AssetProcessor/native/utilities/Builder.cpp

```diff
@@ -0,0 +1,370 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <QCoreApplication>
+#include <QElapsedTimer>
+#include <AzCore/Settings/CommandLine.h>
+#include <AzCore/Settings/SettingsRegistry.h>
+#include <AzCore/Settings/SettingsRegistryMergeUtils.h>
+#include <AzCore/std/parallel/binary_semaphore.h>
+#include <AzCore/Utils/Utils.h>
+#include <utilities/Builder.h>
+#include <utilities/AssetBuilderInfo.h>
+
+namespace AssetProcessor
+{
+    //! Amount of time in milliseconds to wait between checking the status of the AssetBuilder process and pumping the stdout/err pipes
+    constexpr int MaximumSleepTimeMs = 10;
+    constexpr int MillisecondsInASecond = 1000;
+    constexpr const char* BuildersFolderName = "Builders";
+
+    bool Builder::IsConnected() const
+    {
+        return m_connectionId > 0;
+    }
+
+    AZ::Outcome<void, AZStd::string> Builder::WaitForConnection()
+    {
+        if (m_startupWaitTimeS == 0)
+        {
+            const auto* settingsRegistry = AZ::SettingsRegistry::Get();
+            if (settingsRegistry)
+            {
+                settingsRegistry->Get(m_startupWaitTimeS, "/Amazon/AssetProcessor/Settings/BuilderManager/StartupTimeoutSeconds");
+            }
+        }
+
+        if (m_connectionId == 0)
+        {
+            bool result = false;
+            QElapsedTimer ticker;
+
+            ticker.start();
+
+            while (!result)
+            {
+                result = m_connectionEvent.try_acquire_for(AZStd::chrono::milliseconds(MaximumSleepTimeMs));
+
+                PumpCommunicator();
+
+                if (ticker.elapsed() > m_startupWaitTimeS * MillisecondsInASecond || m_quitListener.WasQuitRequested() || !IsRunning())
```

Comment:
What is AzTime?  Searching the codebase for that isn't coming up with anything.  Do you mean AZStd::chrono?  That might be a bit of a pain to use here since .elapsed() returns a qint64 in milliseconds